tests/TranslatableTest.php

```diff
@@ -634,4 +634,20 @@ public function test_it_uses_fallback_locale_if_default_is_empty()
         $this->app->setLocale('fr');
         $this->assertEquals('Tunisia', $country->name);
     }
+
+    public function test_translation_with_multiconnection()
+    {
+        $country = new Country();
+        $country->setConnection('mysql2');
+        $country->code = 'sg';
+        $country->{'name:sg'} = 'Singapore';
+        $this->assertTrue($country->save());
+
+        $countryId = $country->id;
+
+        $country = new Country();
+        $country->setConnection('mysql2');
+        $sgCountry = $country->find($countryId);
+        $this->assertEquals($sgCountry->translate('sg')->name, 'Singapore');
+    }
```

Comment:
@dimsav Noted & added more assertions as you requested and also I added a empty check before `setConnection`